src/datatypes/src/types/decimal_type.rs

```diff
@@ -46,7 +57,8 @@ impl Decimal128Type {
 
 impl DataType for Decimal128Type {
     fn name(&self) -> &str {
-        "decimal128"
+        // TODO(QuenKar): support precision and scale information in name
```

Comment:
Why not add them to the name currently?

Reply:
we can't return a `&str` with precision and scale info in runtime.

Reply:
Yes, so we can keep the name as a field when creating the `Decimal128Type` and return it directly.

Reply:
After discuss with @evenyag , we decide to refactor this method because of some Nested type with extra info may meet the same question.